test/codeu/chat/server/ConversationAccessServerTest.java

```diff
@@ -0,0 +1,535 @@
+ *  - codeu.chat.server.Controller.getPermissionMap(ConversationHeader conversation)
+ *  - codeu.chat.server.Controller.updatePermissionMap(Uuid id, HashMap<Uuid, byte> accessMap)
+ *  - codeu.chat.server.Controller.checkMembership(Uuid id, ConversationHeader conversation)
+ *  - ...server.Controller.demoteOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.ban(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addMember(byte access, Uuid target, ConversationHeader conversation)
+ *       
+ */
+
+
+package codeu.chat.server;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.common.BasicController;
+import codeu.chat.common.BasicView;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+
+public final class ConversationAccessServerTest {
+
+  private Model model;
+  private BasicController controller;
+  private BasicView view;
+  private byte creatorByte = 0b111; // Before testing, this must be the equivalent of 111.
+  private byte ownerByte = 0b011; // Before testing, this must be the equivalent of 011.
+  private byte memberByte = 0b001; // Before testing, this must be the equivalent of 001.
+  private byte trollByte = 0b000; // Before testing, this must be the equivalent of 000.
+  
+  @Before
+  public void doBefore() {
+    model = new Model();
+    controller = new Controller(Uuid.NULL, model);
+    view = new View(model);			
+  }
+
+  @Test
+  public void testAddConversation() {
+
+    final User user = controller.newUser("user");
+
+    assertNotNull(
+        "Check that user has a valid reference",
+        user);
+
+    final ConversationHeader conversation = controller.newConversation(
+        "conversation",
+        user.id);
+
+    assertFalse(
+        "Check that conversation has a valid reference",
+        conversation == null);    
+        
+    assertNotNull(
+         "Check that user has a conversation access map",
+          controller.getPermissionMap(conversation));
+  }
+  
+  @Test
+  public void testGetAndUpdateAccess(){
+    final User creator = controller.newUser("creator");
+    final User user = controller.newUser("user");
+     
+    ConversationHeader conversation = controller.newConversation("chat", creator.id);
+    HashMap<Uuid, Byte> accessMap = controller.getPermissionMap(conversation);
+    
+    byte memberByte = 0b001; // Before testing, this must be the equivalent of 001.
+    
+    accessMap.put(user.id, memberByte);
+    controller.updatePermissionMap(conversation, accessMap);
+    
+    assertTrue(
+        "Check that the allowances can be updated",
+        controller.getPermissionMap(conversation).size()== 1);
```

Comment:
Again, you should go a step further, and examine the contents of the permission map. Here, you're testing that the permission map has an additional entry, but you should also make sure that the new entry is correct.